server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -575,35 +575,55 @@ void refreshPoolsForMasterPool(Pool pool, boolean updateStackDerived, boolean la
         regenerateCertificatesByEntIds(updatedMasterPools, lazy);
     }
 
-    @Transactional
+    /**
+     * Deletes all known expired pools. The deletion of expired pools also triggers entitlement
+     * revocation and consumer compliance recalculation.
+     * <p></p>
+     * This method will delete pools in blocks, using a new transaction for each block unless a
+     * transaction was already started before this method is called.
+     */
     public void cleanupExpiredPools() {
         int count = 0;
         boolean loop = false;
 
-        do {
-            List<Pool> pools = poolCurator.listExpiredPools(PoolCurator.EXPIRED_POOL_BLOCK_SIZE);
-            count += pools.size();
-
-            // TODO: Do we still need this? This bit is incredibly slow.
-            for (Iterator<Pool> pi = pools.iterator(); pi.hasNext();) {
-                Pool pool = pi.next();
-                log.info("Cleaning up expired pool: {} (expired: {})", pool.getId(), pool.getEndDate());
-            }
+        log.debug("Beginning cleanup expired pools job");
 
-            // Delete the block of pools
-            this.deletePools(pools);
-
-            // Cleanup memory to try to avoid flooding the heap
-            this.poolCurator.clear();
+        do {
+            // This call is run within a new transaction if we're not already in a transaction
+            int blockSize = this.cleanupExpiredPoolsImpl();
+            count += blockSize;
 
-            loop = pools.size() >= PoolCurator.EXPIRED_POOL_BLOCK_SIZE;
+            loop = blockSize >= PoolCurator.EXPIRED_POOL_BLOCK_SIZE;
         } while (loop);
 
         if (count > 0) {
             log.info("Cleaned up {} expired pools", count);
         }
     }
 
+    /**
+     * Performs the cleanup of a block of expired pools.
+     *
+     * @return
+     *  the number of expired pools deleted as a result of this method
+     */
+    @Transactional
```

Comment:
Since cleanupExpiredPoolsImpl is private, the Transactional attribute doesn't have any effect. Thats why @vritant  saw TransactionRequiredException

Either make the method public or dont flush. There is actually no need to flush since deletePools is gonna flush.